TransactionsLib.js

```diff
@@ -0,0 +1,93 @@
+const moment = require('moment');
+const loggerMessages = require('./loggerMessages');
+const moneyConverter = require('./moneyConverter');
+const fs = require('fs');
+
+class Transaction {
+    constructor(transactionDetails, logger) {
+
+        if (transactionDetails.length !== 5) {
+            loggerMessages.logFatal('Wrong number of parameters for Transaction constructor\n' +
+                                `(have ${transactionDetails.length} instead of 5)`, logger)
+        }
+
+        const [dateString, from, to, narrative, amountStrig] = transactionDetails;
+
+        this.date = parseTime(dateString, logger);
+        this.from = from;
+        this.to = to;
+        this.narrative = narrative;
+        this.ammount = moneyConverter.convertMoneyStringToInt(amountStrig, logger);
+    
+    }
+
+}
+
+function parseTime(dateString, logger) {
+    
+    // CVS format
+    date = moment(dateString, 'DD/MM/YYYY');

```

Comment:
This date variable isn't declared, and if it's valid at the first check then it's never returned.